src/Compilers/Core/Portable/Symbols/SymbolEqualityComparer.cs

```diff
@@ -25,9 +25,13 @@ public sealed class SymbolEqualityComparer : IEqualityComparer<ISymbol?>
         /// </summary>
         public static readonly SymbolEqualityComparer IncludeNullability = new SymbolEqualityComparer(TypeCompareKind.ConsiderEverything2); //TODO: should this be explicitly *not* compare everything
 
+        /// <summary>
+        /// Compares two <see cref="ISymbol"/> instances, ignoring all options
```

Comment:
>ignoring all options [](start = 58, length = 20)

What all options, I don't think we advertise what can be ignored. We say that nullability is ignored by default and should be added explicitly.

Reply:
Based on the changes in this PR, I see no reasons making this API public. Am I missing something?

---
In reply to: [493926708](https://github.com/dotnet/roslyn/pull/47960#discussion_r493926708) [](ancestors = 493926708)

Reply:
`FindValidCloneMethod` calls SymbolEqualityComparer.IgnoreAll. The method is linked on the IDE side, and the IDE does not have access to SymbolEqualityComparer.IgnoreAll, so we need to expose it here.

Reply:
> and the IDE does not have access to SymbolEqualityComparer.IgnoreAll, so we need to expose it here.

I see. I am still not comfortable exposing it without making it clear what it means.  

---
In reply to: [493940577](https://github.com/dotnet/roslyn/pull/47960#discussion_r493940577) [](ancestors = 493940577)

Reply:
I would prefer making FindValidCloneMethod a public extension method, or better adding IsRecord property to ITypeSymbol and keeping things internal and where they were.  

---
In reply to: [493951181](https://github.com/dotnet/roslyn/pull/47960#discussion_r493951181) [](ancestors = 493951181,493940577)

Reply:
See [https://github.com/dotnet/roslyn/issues/47713](https://github.com/dotnet/roslyn/issues/47713)

---
In reply to: [493953924](https://github.com/dotnet/roslyn/pull/47960#discussion_r493953924) [](ancestors = 493953924,493951181,493940577)

Reply:
@jcouv can you please assist with this part of the PR?  Thanks! :)

Reply:
@AlekseyTs I have added IsRecord as a property in ITypeSymbol, which hopefully gets rid of most of your previously mentioned concerns.